barretenberg/cpp/src/barretenberg/vm2/testing/fixtures.cpp

```diff
@@ -169,39 +171,19 @@ ContractInstance random_contract_instance()
 
 std::pair<tracegen::TraceContainer, PublicInputs> get_minimal_trace_with_pi()
 {
-    AvmTraceGenHelper trace_gen_helper;
+    // cwd is expected to be barretenberg/cpp/build.
+    auto data = read_file("../src/barretenberg/vm2/common/minimal_tx.testdata.bin");
+    AvmProvingInputs inputs = AvmProvingInputs::from(data);
 
-    PublicInputs public_inputs = {
-        .globalVariables = {
-            .blockNumber = 42,
-        },
-        .startTreeSnapshots = {
-            .l1ToL2MessageTree = {
-                .root = 111,
-                .nextAvailableLeafIndex = 222,
-            },
-            .noteHashTree = {
-                .root = 333,
-                .nextAvailableLeafIndex = 444,
-            },
-            .nullifierTree = {
-                .root = 555,
-                .nextAvailableLeafIndex = 666,
-            },
-            .publicDataTree = {
-                .root = 777,
-                .nextAvailableLeafIndex = 888,
-            },
-        },
-        .reverted = false,
-    };
+    AvmSimulationHelper simulation_helper(inputs.hints);
+
+    auto events = simulation_helper.simulate();
+
+    AvmTraceGenHelper trace_gen_helper;
 
-    auto trace = trace_gen_helper.generate_trace({
-            .alu = { { .operation = simulation::AluOperation::ADD, .a = MemoryValue::from<uint16_t>(1), .b = MemoryValue::from<uint16_t>(2), .c = MemoryValue::from<uint16_t>(3) }, },
-        },
-        public_inputs);
+    auto trace = trace_gen_helper.generate_trace(std::move(events), inputs.publicInputs);
```

Comment:
Just consider doing this only once. You can make the trace variable `static` and a lambda that calls itself.